### PR TITLE
Add vengi

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/mgerhardy/vengi/issues/175
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://github.com/mgerhardy/vengi/pull/179

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build io.github.mgerhardy.vengi

File: io.github.mgerhardy.vengi.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    download-if: active-gl-driver
+    enable-if: active-gl-driver
+
+modules:
+  - name: essential-dirs
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib/GL
+
+  - name: opencl-headers
```
Comment (barthalion) [inline io.github.mgerhardy.vengi.yml:30]: These files should be cleaned up.

File: io.github.mgerhardy.vengi.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+rename-icon: voxedit-icon
+rename-desktop-file: vengi-voxedit.desktop
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
+  - --filesystem=home
+
+add-extensions:
+  org.freedesktop.Platform.GL:
```
Comment (barthalion) [inline io.github.mgerhardy.vengi.yml:15]: What is this needed for? GL extension is enabled automatically.

**Comment by Mailaender** [comment]
bot, build io.github.mgerhardy.vengi

File: io.github.mgerhardy.vengi.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  - name: vengi
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DUNITTESTS=OFF
+      - -DVISUALTESTS=OFF
+      - -DGAMES=OFF
+      - -DMAPVIEW=OFF
+      - -DAIDEBUG=OFF
+      - -DVOXCONVERT=OFF
+      - -DTHUMBNAILER=OFF
```
Comment (Newbytee) [inline io.github.mgerhardy.vengi.yml:46]: Just curious, why are these being disabled?

File: io.github.mgerhardy.vengi.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: 22.08
+sdk: org.freedesktop.Sdk
+command: vengi-voxedit
+rename-icon: voxedit-icon
+rename-desktop-file: vengi-voxedit.desktop
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
```
Comment (Newbytee) [inline io.github.mgerhardy.vengi.yml:11]: This seems to use SDL 2, are you sure Wayland doesn't work?

File: io.github.mgerhardy.vengi.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  - --filesystem=home
+  cleanup:
```
Comment (hfiguiere) [inline io.github.mgerhardy.vengi.yml:13]:
```suggestion
cleanup:
```

it's yaml

File: io.github.mgerhardy.vengi.yml (hunk 4 above)
Comment (Mailaender) [reply io.github.mgerhardy.vengi.yml:11]: Are you sure it works?

Comment (Newbytee) [reply io.github.mgerhardy.vengi.yml:11]: No, that's why I'm asking. I can try it.

File: io.github.mgerhardy.vengi.yml (hunk 3 above)
Comment (Mailaender) [reply io.github.mgerhardy.vengi.yml:46]: Because it won't work inside the sandbox environment.

**Comment by Mailaender** [comment]
bot, build io.github.mgerhardy.vengi.voxedit

File: io.github.mgerhardy.vengi.voxedit.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
+app-id: io.github.mgerhardy.vengi.voxedit
```
Comment (hfiguiere) [inline io.github.mgerhardy.vengi.voxedit.yml:1]: shouldn't this just be `io.github.mgerhardy.vengi`?

Comment (Mailaender) [reply io.github.mgerhardy.vengi.voxedit.yml:1]: There are more tools https://github.com/mgerhardy/vengi/pull/179#discussion_r963615244

**Comment by Mailaender** [comment]
bot, build io.github.mgerhardy.vengi.voxedit

File: io.github.mgerhardy.vengi.voxedit.yml (hunk 6 above)
Comment (hfiguiere) [reply io.github.mgerhardy.vengi.voxedit.yml:1]: if they are build from the same source, they can be exported from the same flatpak. And have each their own desktop entry

File: io.github.mgerhardy.vengi.yml (hunk 4 above)
Comment (Newbytee) [reply io.github.mgerhardy.vengi.yml]: Wayland seems to work fine if I enable the Wayland socket and set X11 to be a fallback.

Comment (Mailaender) [reply io.github.mgerhardy.vengi.yml]: Okay, thanks.

**Comment by barthalion** [comment]
bot, build io.github.mgerhardy.vengi.voxedit

**Comment by Mailaender** [comment]
bot, build io.github.mgerhardy.vengi.voxedit

**Pull request closed without merging**